README.md

```diff
@@ -1 +1,510 @@
+          </label>
+        )
+      })}
+      {inputMethods.getError() && <p>This is an error!</p>}
+    </FormContext>
+  )
+}
+```
+
+### useSelect(path)
+
+##### Description
+
+Use this hook to build a select field in your form.
+
+##### Parameters:
+
+- `path`: String which represents the path to the data field of the JSON Schema that this input will be built for.
+
+##### Return:
+
+Returns an object with the following fields, besides the common one's:
+
+- `getLabelProps()`: use this with the spread operator inside a `<label>` tag and get the benefit of having all the important fields of the label filled in for you and the associated select (the `for` property) with it.
+- `getItems()`: use this to get all the values that are possible to be in the radio buttons
+- `getItemOptionProps(index)`: use this with the spread operator inside an `<option>` tag and get the benefit of the validator, id field and name for the item in the specified index from `getItems()`
+- `getSelectProps()`: use this with the spread operator inside a `<select>` tag to get validation and register it with the react-hook-form-jsonschema.
+
+##### Example:
+
+```JSX
+function InputField(props) {
+  const inputMethods = useSelect('#/some/child/with/limited/possible/values');
+
+  return (
+    <FormContext schema={personSchema}>
+      <label {...inputMethods.getLabelProps()}>{inputMethods.name}</label>
+      <select {...inputMethods.getSelectProps()}>
+        {inputMethods.getItems().map((value, index) => {
+          return (
+            <option
+              {...inputMethods.getItemOptionProps(index)}
+              key={`${value}${index}`}
+            >
+              {value}
+            </option>
+          )
+        })}
+      </select>
+      {inputMethods.getError() && <p>This is an error!</p>}
+    </FormContext>
+  )
+}
+```
+
+### useTextArea(path)
+
+##### Description
+
+Use this hook to build a textarea field in the form.
+
+##### Parameters:
+
+- `path`: String which represents the path to the data field of the JSON Schema that this input will be built for.
+
+##### Return:
+
+Returns an object with the following fields, besides the common one's:
+
+- `getLabelProps()`: use this with the spread operator inside a `<label>` tag and get the benefit of having all the important fields of the label filled in for you and the associated input (the `for` property) with it.
+- `getTextAreaProps()`: use this with the spread operator inside an `<textarea>` tag and get the benefit of the validator, id field, name and an associated label with it
+
+##### Example:
+
+```JSX
+function HiddenField(props) {
+  const inputMethods = useTextArea('#/some/child/you/want/as/TextArea');
+
+  return (
+    <FormContext schema={personSchema}>
+      <label {...inputMethods.getLabelProps()}>
+        {inputMethods.name}
+      </label>
+      <textarea {...inputMethods.getTextAreaProps()} />
+    </FormContext>
+  )
+}
+```
```

Comment:
It would be nice to have a "how to contribute"section, but you can do this later!